### PR TITLE
Fixed bug with entities order in files tech

### LINKED ISSUE #129: Files tech not keep order of entities
Files must be sorted first by entity and then by tech.

**Example**

`?.deps.js`

```js
exports.deps = [
    { block: 'block-1' },
    { block: 'block-2' }
];
```

In `?.files` target will be files in the following order:

```
file-1.ext-1
file-2.ext-1
file-1.ext-2
file-2.ext-2
```

Expected order:
```
file-1.ext-1
file-1.ext-2
file-2.ext-1
file-2.ext-2
```

### PR DESCRIPTION
**Pull request by blond**
Resolved #129

### CODE REVIEW

File: techs/files.js [hunk 1]
```diff
@@ -106,39 +106,42 @@ module.exports = inherit(require('enb/lib/tech/base-tech.js'), {
...
 
-                            addEntityFiles(entities);
+                            addEntities(entities);
                         }
 
-                        fileList.addFiles(Object.keys(files).map(function (filename) {
-                            return files[filename];
-                        }));
+                        function addEntities(entities) {
+                            fileList.addFiles(entities.files.filter(function (file) {
```
Comment (tadatuta) [inline techs/files.js]: isn't it possible to store files and dirs in the same hash and as so reuse filter callback for both cases?

**Comment by tadatuta** [comment]
lgtm

File: techs/files.js (hunk 1 above)
Comment (blond) [inline techs/files.js]: Nice idea! It has been fixed.

**Comment by tadatuta** [comment]
lgtm

File: test/techs/files.test.js [hunk 2]
```diff
@@ -70,6 +70,51 @@ describe('techs', function () {
                 .then(done, done);
         });
 
+        it('must keep order of entities', function (done) {
+            mockFs({
+                blocks: {
```
Comment (aliosv) [inline test/techs/files.test.js]: TODO: добавить еще один уровень.

**Pull request merged**